trin-execution/src/main.rs

```diff
@@ -44,27 +39,33 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
             .expect("signal ctrl_c should never fail");
     });
 
-    for block_number in starting_block_number..=get_spec_block_number(SpecId::MERGE) {
+    let mut block_number = state.next_block_number();
+
+    let end_block = get_spec_block_number(SpecId::MERGE);
+    while block_number < end_block {
         if rx.try_recv().is_ok() {
             state.database.db.flush()?;
             info!(
-                "Received SIGINT, stopping execution: {block_number} {}",
+                "Received SIGINT, stopping execution: {} {}",
+                block_number - 1,
                 state.get_root()?
             );
             break;
         }
 
-        let timer = start_timer_vec(&BLOCK_PROCESSING_TIMES, &["fetching_block_from_era"]);
-        let block = era_manager.get_next_block().await?;
-        stop_timer(timer);
-        let timer = start_timer_vec(&BLOCK_PROCESSING_TIMES, &["processing_block"]);
-        if block.header.number == 0 {
+        let end =
```

Comment:
> Can you elaborate on this logic.
> 
> * What's wrong with having fixed number of blocks to process (could even be configurable with a flag).
> * What are we trying to accomplish and why?
> * Why use 86 (what is special about it)?

I think it was a sweet, spot for getting over the dao fork, but since we have the early exit code now, I can remove this